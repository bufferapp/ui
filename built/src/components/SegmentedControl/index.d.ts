export { default } from './SegmentedControl';
//# sourceMappingURL=index.d.ts.map