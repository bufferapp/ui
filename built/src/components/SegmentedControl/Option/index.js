export { default } from './Option';
//# sourceMappingURL=index.js.map