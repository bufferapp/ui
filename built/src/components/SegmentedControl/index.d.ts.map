{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../src/components/SegmentedControl/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,oBAAoB,CAAA"}