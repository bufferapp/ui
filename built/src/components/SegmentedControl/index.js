export { default } from './SegmentedControl';
//# sourceMappingURL=index.js.map