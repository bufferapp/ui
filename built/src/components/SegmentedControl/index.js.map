{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../../src/components/SegmentedControl/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,oBAAoB,CAAA","sourcesContent":["export { default } from './SegmentedControl'\n"]}