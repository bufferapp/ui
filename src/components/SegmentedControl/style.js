import styled from 'styled-components'
import { gray, white } from '../style/colors'
import { borderRadius } from '../style/borders'

export const Container = styled.div`
  display: flex;
  border: 1px solid ${gray};
  border-radius: ${borderRadius};
  padding: 3px;
  background-color: ${white};
`
