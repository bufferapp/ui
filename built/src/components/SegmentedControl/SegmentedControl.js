import React from 'react';
import PropTypes from 'prop-types';
import Option from './Option';
import { Container } from './style';
const SegmentedControl = ({ children }) => (React.createElement(Container, { role: "group" }, children));
SegmentedControl.propTypes = {
    children: PropTypes.node.isRequired,
};
SegmentedControl.defaultProps = {};
SegmentedControl.Option = Option;
export default SegmentedControl;
//# sourceMappingURL=SegmentedControl.js.map