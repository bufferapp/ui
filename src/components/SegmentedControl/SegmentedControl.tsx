import React from 'react'
import PropTypes from 'prop-types'

import Option from './Option'
import { Container } from './style'

/**
 * Segmented control used for quickly selecting between a small set of mutually exclusive options.
 * They work like a group of radio inputs in that no more than one may be selected at a time.
 * Segmented controls are useful for scenarios where there are only a few options.
 * If you need to present many options, consider using another component such as a Select.
 */
// @ts-expect-error TS(7031) FIXME: Binding element 'children' implicitly has an 'any'... Remove this comment to see the full error message
const SegmentedControl = ({ children }) => (
  <Container role="group">{children}</Container>
)

SegmentedControl.propTypes = {
  children: PropTypes.node.isRequired,
}

SegmentedControl.defaultProps = {}

SegmentedControl.Option = Option

export default SegmentedControl
