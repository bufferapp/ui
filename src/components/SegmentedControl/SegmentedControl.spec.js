import snap from 'jest-auto-snapshots'
import SegmentedControl from './SegmentedControl'

snap(SegmentedControl, './SegmentedControl.jsx')
