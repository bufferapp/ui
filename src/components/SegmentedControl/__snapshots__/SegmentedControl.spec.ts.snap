// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > SegmentedControl Matches snapshot when passed only required props 1`] = `
<styled.div
  role="group"
>
  <NodeFixture />
</styled.div>
`;
